File: 008/Ruby/problem008_test.rb
require "test/unit"

require "problem008"

class TestProblem008 < Test::Unit::TestCase

  def test_largest_five_digit_product
    assert_equal 0, 1234.largest_five_digit_product
    assert_equal 1890, 23579.largest_five_digit_product
    assert_equal 1890, 123579.largest_five_digit_product
    assert_equal 1890, 235791.largest_five_digit_product
    assert_equal 3125, 223355555337.largest_five_digit_product
    assert_equal 1890, 123223957112110.largest_five_digit_product
    assert_equal 32, 1111222221111113.largest_five_digit_product
  end
  
end

File: 008/Haskell/Problem008Test.hs
module Main where

import Test.HUnit
import System.Exit
import Problem008

largestFiveDigitProductTests = TestList [
  TestCase (0 @=? largestFiveDigitProduct 1234),
  TestCase (1890 @=? largestFiveDigitProduct 23579),
  TestCase (1890 @=? largestFiveDigitProduct 123579),
  TestCase (1890 @=? largestFiveDigitProduct 235791),
  TestCase (3125 @=? largestFiveDigitProduct 223355555337),
  TestCase (1890 @=? largestFiveDigitProduct 123223957112110),
  TestCase (32 @=? largestFiveDigitProduct 1111222221111113),
  TestCase (1 @=? 1)
  ]

tests = TestList [
	TestLabel "testing largestFiveDigitProduct" largestFiveDigitProductTests,
	TestCase (1 @=? 1)
	]

main = runTestTT tests >>= checkTests
  where
    checkTests counts = case counts of
    	Counts _ _ errors failures ->
    		if (errors + failures > 0)
    		then exitFailure
    		else (exitWith ExitSuccess)
